### ISSUE TITLE
No sound on Fedora 34 with Pipewire

### ISSUE THREAD

**Issue by term1n4l** [kthakore/frozen-bubble#75]
Fedora 34 is now using Pipewire as it's default sound server. When launching frozen-bubble 2.2.1-beta1 (current version in the repos) (x86_64) It crashes with the following error (if launched from bash):

[Sound Init] Can't use an undefined value as a SCALAR reference at /usr/bin/frozen-bubble line 338.

so it looks like a variable is not being set properly in this configuration. I can only launch frozen bubble if I specify `--no-sound`. This is a clean install of Fedora 34 KDE edition running under wayland by default.

pipewire is version 0.26.3 from the fedora repos (possibly updates-testing). note I have installed pulseaudio-utils as well to ensure I have the pulseaudio utilities installed in case this was the issue.

**Comment by term1n4l** [comment]
After further investigation I discovered SDL2 should provide support for pipewire based on:

https://www.phoronix.com/scan.php?page=news_item&px=SDL2-Lands-PipeWire-Audio

I can't see any SDL2_Pango in the fedora repos though. After installing the SDL2 libraries I no longer crash out, but I still have no sound when running without frozen-bubble directly

**Comment by romulasry** [comment]
Is this fixed now?

**Comment by romulasry** [comment]
Works for me on rawhide.

**Comment by term1n4l** [comment]
This is not working for me on Fedora 35 on the same system I had this issue initially. I have some SDL2 libraries installed and I have the SDL12-compat library as well. I haven't tested with a clean install recently though. I did notice that a Fedora system I've upgraded from F33 (or earlier) has frozen bubble still working with sound on F34, I recently upgraded that same host to Fedora 35 but I haven't tested frozen bubble yet.

**Comment by term1n4l** [comment]
So after further troubleshooting (comparison with a brand new system etc) I removed all the SDL2_* packages I had installed previously in my attempts to get this working on F34. I was missing a library but it was libffado which doesn't appear to be relevant to my system anyway, I also noted that jack-audio-connection-kit was being pulled on the clean system, but I am using pipewire-jack-audio-connection-kit. After I removed the SDL2_ libraries I began noticing graphical issues in launching the game. I decided to remove my user profile settings for frozen-bubble (e.g. ~/.frozen-bubble) and this resolved both the graphical issue and the sound issue for me. I now have working sound in this game again!
